src/sources/socket/mod.rs

```diff
@@ -202,23 +212,114 @@ impl SourceConfig for SocketConfig {
+
+                unix::unix_stream(config, decoder, cx.shutdown, cx.out, log_namespace)
             }
         }
     }
 
-    fn outputs(&self, _global_log_namespace: LogNamespace) -> Vec<Output> {
-        vec![Output::default(self.output_type())]
+    fn outputs(&self, global_log_namespace: LogNamespace) -> Vec<Output> {
+        let log_namespace = global_log_namespace.merge(Some(self.log_namespace()));
+
+        let schema_definition = self
+            .decoding()
+            .schema_definition(log_namespace)
+            .with_standard_vector_source_metadata();
+
+        let schema_definition = match &self.mode {
+            Mode::Tcp(config) => {
+                let host_key_path = config
+                    .host_key()
+                    .as_ref()
+                    .map(|x| owned_value_path!(x))
+                    .map(LegacyKey::InsertIfEmpty);
+
+                let port_key_path = config
+                    .port_key()
+                    .as_ref()
+                    .map(|x| owned_value_path!(x))
+                    .map(LegacyKey::InsertIfEmpty);
+
+                schema_definition
+                    .with_source_metadata(
+                        Self::NAME,
+                        host_key_path,
+                        &owned_value_path!(log_schema().host_key()),
+                        Kind::bytes(),
+                        None,
+                    )
+                    .with_source_metadata(
+                        Self::NAME,
+                        port_key_path,
+                        &owned_value_path!("port"),
+                        Kind::bytes(),
+                        None,
+                    )
+            }
+            Mode::Udp(config) => {
+                let host_key_path = config
+                    .host_key()
+                    .as_ref()
+                    .map(|x| owned_value_path!(x))
+                    .map(LegacyKey::InsertIfEmpty);
+
+                let port_key_path = config
+                    .port_key()
+                    .as_ref()
+                    .map(|x| owned_value_path!(x))
+                    .map(LegacyKey::InsertIfEmpty);
+
+                schema_definition
+                    .with_source_metadata(
+                        Self::NAME,
+                        host_key_path,
+                        &owned_value_path!(log_schema().host_key()),
+                        Kind::bytes(),
+                        None,
+                    )
+                    .with_source_metadata(
+                        Self::NAME,
+                        port_key_path,
+                        &owned_value_path!("port"),
+                        Kind::bytes(),
+                        None,
+                    )
+            }
+            Mode::UnixDatagram(config) => {
+                let host_key_path = config
+                    .host_key()
+                    .as_ref()
+                    .map(|x| owned_value_path!(x))
```

Comment:
The path macros (such as `owned_value_path`) can only be used if `host_key_path` is known to always be a single segment (which it's not). You can use `parse_value_path` to parse a full path instead. This returns a `Result`, which you can call `.ok()` on to convert to an `Option`, and use that with `.and_then` on the existing optional path to merge both Options into a single one. This applies to all of the user-defined paths here.